### PR TITLE
chore: Creates TF models & interfaces for new `mongodbatlas_encryption_at_rest_private_endpoint` data source

### PR DESCRIPTION
**Pull request by maastha**
## Description

Creates TF models & interfaces for new mongodbatlas_encryption_at_rest_private_endpoint data source:

Adds:

- TF model
- TF schema
- CRUD skeleton handlers
- updates atlas dev-preview Go SDK

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,7 +24,7 @@ require (
 	github.com/zclconf/go-cty v1.15.0
 	go.mongodb.org/atlas v0.36.0
 	go.mongodb.org/atlas-sdk/v20240530005 v20240530005.0.0
-	go.mongodb.org/atlas-sdk/v20240805001 v20240805001.0.1-0.20240809201355-5fc86d455f4c
+	go.mongodb.org/atlas-sdk/v20240805001 v20240805001.0.1-0.20240813143344-1cfe40386cbc
```
Comment (maastha) [inline go.mod:27]: updating SDK preview version to pick up latest changes in API spec

File: internal/service/encryptionatrestprivateendpoint/tfplugingen/generator_config.yml [hunk 2]
```diff
@@ -13,3 +13,12 @@ resources:
       path: /api/atlas/v2/groups/{groupId}/encryptionAtRest/{cloudProvider}/privateEndpoints/{endpointId}
       method: DELETE
 
+data_sources:
+  encryption_at_rest_private_endpoint:
+    read:
+      path: /api/atlas/v2/groups/{groupId}/encryptionAtRest/{cloudProvider}/privateEndpoints/{endpointId}
+      method: GET
+  # encryption_at_rest_private_endpoints:
```
Comment (maastha) [inline internal/service/encryptionatrestprivateendpoint/tfplugingen/generator_config.yml:21]: will uncomment in follow-up PR for adding plural data source

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: internal/service/encryptionatrestprivateendpoint/data_source_schema.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+func DataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"cloud_provider": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the cloud provider of the private endpoint.",
```
Comment (marcosuma) [inline internal/service/encryptionatrestprivateendpoint/data_source_schema.go:14]: just wondering: will updating these later be a manual operation?

**Pull request merged**